File: nopCommerce_4.30/Libraries/Mall.DTO/Settlement/PendingSettlementOrders.cs
using System;

namespace Mall.DTO
{
    /// <summary>
    /// 待结算订单
    /// </summary>
    public class PendingSettlementOrders
    {
        /// <summary>
        /// 订单ID
        /// </summary>
        public long OrderId { set; get; }

        /// <summary>
        /// 订单号字符串
        /// </summary>
        public string strOrderId { get { return OrderId.ToString(); } }

        /// <summary>
        /// 订单状态(目前只针对已完成订单)
        /// </summary>
        public string Status
        {
            set;
            get;
        }

        /// <summary>
        /// 店铺名称
        /// </summary>
        public string ShopName { set; get; }

        /// <summary>
        /// 支付方式
        /// </summary>

        public string PaymentTypeName { set; get; }

        /// <summary>
        /// 订单金额
        /// </summary>
        public decimal OrderAmount { get; set; }

        /// <summary>
        /// 平台佣金
        /// </summary>
        public decimal PlatCommission { set; get; }

        /// <summary>
        /// 分销佣金
        /// </summary>
        public decimal DistributorCommission { set; get; }


        /// <summary>
        /// 平台佣金退款
        /// </summary>
        public decimal PlatCommissionReturn { set; get; }


        /// <summary>
        /// 分销佣金退款
        /// </summary>
        public decimal DistributorCommissionReturn { set; get; }

        /// <summary>
        /// 运费
        /// </summary>
        public decimal FreightAmount { get; set; }

        /// <summary>
        /// 退款金额
        /// </summary>
        public decimal RefundAmount { set; get; }

        /// <summary>
        /// 商家结算金额
        /// </summary>
        public decimal SettlementAmount { set; get; }

        /// <summary>
        /// 订单完成时间
        /// </summary>
        /// 
      
        public DateTime OrderFinshTime { set; get; }

        public string OrderFinshTimeStr { get { return OrderFinshTime.ToString("yyyy-MM-dd HH:mm:ss"); } }
        public decimal TaxAmount
        {
            get;
            set;
        }
        
        /// <summary>
        /// 积分抵扣金额
        /// </summary>
        
        public decimal IntegralDiscount
        {
            get;
            set;
        }
        /// <summary>
		/// 退款时间
		/// </summary>
        public DateTime? RefundDate
        {
            get;set;
        }
        /// <summary>
		/// 创建时间
		/// </summary>
        public DateTime CreateDate
        { get; set; }
        /// <summary>
		/// 退款时间
		/// </summary>
        public string RefundDateStr { get { return RefundDate.HasValue ? RefundDate.Value.ToString("yyyy-MM-dd HH:mm:ss") : string.Empty; } }
        /// <summary>
		/// 创建时间
		/// </summary>
        public string CreateDateStr { get { return CreateDate.ToString("yyyy-MM-dd HH:mm:ss"); } }
        /// <summary>
        /// 结算周期范围(不在当前结算周期内的)
        /// </summary>
        public string SettlementCycle
        {
            set
            {
                if (value != null)  _SettlementCycle = value;
            }

            get { return _SettlementCycle; }
        }
        private string _SettlementCycle = "";
    }
}

File: nopCommerce_4.30/Presentation/Mall.Web/bin/Debug/netcoreapp3.1/win-x64/Areas/Admin/Scripts/PendingSettlementOrders.js
var datacols = [[
             {
             	field: "OrderId", title: '订单号', width: 'auto',
             	formatter: function (value, row, index) {
             		return '<a title="' + row.SettlementCycle + '" href="/Admin/order/Detail/' + value + '">' + value + '</a>';
             	}
             },

             { field: "Status", title: "订单状态", width: 'auto', align: "center" },
               { field: "ShopName", title: "店铺名称", width: 'auto', align: "center" },
             { field: "OrderAmount", title: "订单实付", width: 'auto', align: "center" },
             { field: "IntegralDiscount", title: "积分抵扣", width: 'auto', align: "center" },
             { field: "PlatCommission", title: "平台佣金", width: 'auto', align: "center" },
             { field: "DistributorCommission", title: "分销佣金", width: 'auto', align: "center" },
             { field: "RefundAmount", title: "退款金额", width: 'auto', align: "center" },
             { field: "SettlementAmount", title: "结算金额", width: 'auto', align: "center" },
             { field: "CreateDateStr", title: "创建时间", width: 'auto', align: "center" },
              { field: "PaymentTypeName", title: "支付方式", width: 'auto', align: "center" },
                {
                	field: "operation", title: "查看详情", width: 'auto', align: "center",

                	formatter: function (value, row, index) {
                		return '<a href="/Admin/Billing/PendingSettlementDetail/' + row.OrderId + '">查看详情</a>';
                	}
                }


]];

var shopId = GetQueryString('shopId');
$(function () {
	//组合显示字段
	//订单表格
	$("#list").MallDatagrid({
		url: 'PendingSettlementOrderList',
		nowrap: false,
		rownumbers: true,
		NoDataMsg: '没有找到符合条件的待结算订单记录',
		border: false,
		fit: true,
		fitColumns: true,
		pagination: true,
		idField: "OrderId",
		pageSize: 15,
		pagePosition: 'bottom',
		operationButtons: "#operationButtons",
		pageNumber: 1,
		queryParams: {shopId:shopId},
		columns: datacols
	});

	$('#searchButton').click(function (e) {
		searchClose(e);
		var startDate = $("#inputStartDate").val();
		var endDate = $("#inputEndDate").val();
		var orderId = $.trim($('#txtOrderId').val());
		var paymentName = $("#payment").val();
		$("#list").MallDatagrid('reload', { startDate: startDate, endDate: endDate, orderId: orderId, paymentName: paymentName, shopId: shopId });
	})
});


$(function () {
	$(".start_datetime").datetimepicker({
		language: 'zh-CN',
		format: 'yyyy-mm-dd',
		autoclose: true,
		weekStart: 1,
		minView: 2
	});
	$(".end_datetime").datetimepicker({
		language: 'zh-CN',
		format: 'yyyy-mm-dd',
		autoclose: true,
		weekStart: 1,
		minView: 2
	});
	//$(".start_datetime").click(function () {
	//    $('.end_datetime').datetimepicker('show');
	//});
	//$(".end_datetime").click(function () {
	//    $('.start_datetime').datetimepicker('show');
	//});
	$('.start_datetime').on('changeDate', function () {
		if ($(".end_datetime").val()) {
			if ($(".start_datetime").val() > $(".end_datetime").val()) {
				$('.end_datetime').val($(".start_datetime").val());
			}
		}

		$('.end_datetime').datetimepicker('setStartDate', $(".start_datetime").val());
	});


});

function ExportExecl() {
	var startDate = $("#inputStartDate").val();
	var endDate = $("#inputEndDate").val();
	var orderId = $.trim($('#txtOrderId').val());
	var paymentName = $("#payment").val();
	var href = $(this).attr('href').split('?')[0] + '?startDate={0}&endDate={1}&shopId={2}&paymentName={3}&orderId={4}'.format(startDate, endDate,shopId, paymentName, orderId);
	$(this).attr('href', href);
}